tests/geospatial/test_zonal_average.py

```diff
@@ -0,0 +1,66 @@
+"""
+This example was adapted from https://github.com/dcherian/dask-demo/blob/main/nwm-aws.ipynb
+"""
+
+import dask
+import flox.xarray
+import numpy as np
+import rioxarray
+import xarray as xr
+
+
+def test_nwm(
+    s3_url,
+    scale,
+    client_factory,
+    cluster_kwargs={
+        "workspace": "dask-engineering",
+        "region": "us-east-1",
+        "wait_for_workers": True,
+    },
+    scale_n_workers={
+        "small": 10,
+        "large": 200,
+    },
+):
+    with client_factory(
+        n_workers=scale_n_workers[scale], **cluster_kwargs
+    ) as client:  # noqa: F841
+        ds = xr.open_zarr(
+            "s3://noaa-nwm-retrospective-2-1-zarr-pds/rtout.zarr", consolidated=True
+        )
+
+        if scale == "small":
+            # 6.03 TiB
+            time_range = slice("2020-01-01", "2020-12-31")
+        else:
+            # 252.30 TiB
+            time_range = slice("1979-02-01", "2020-12-31")
+        subset = ds.zwattablrt.sel(time=time_range)
+
+        with dask.annotate(retries=3):
+            counties = rioxarray.open_rasterio(
+                "s3://nwm-250m-us-counties/Counties_on_250m_grid.tif",
+                chunks="auto",
+            ).squeeze()
+
+        # Remove any small floating point error in coordinate locations
+        _, counties_aligned = xr.align(subset, counties, join="override")
+
+        counties_aligned = counties_aligned.persist()
+
+        county_id = np.unique(counties_aligned.data).compute()
+        county_id = county_id[county_id != 0]
+        print(f"There are {len(county_id)} counties!")
+
+        county_mean = flox.xarray.xarray_reduce(
+            subset,
+            counties_aligned.rename("county"),
+            func="mean",
+            expected_groups=(county_id,),
+        )
+
+        county_mean.load()
+        yearly_mean = county_mean.mean("time")
+        # Save dataset for further analysis
+        yearly_mean.to_netcdf(s3_url)
```

Comment:
This line is currently failing with 

```
self = <scipy.io._netcdf.netcdf_file object at 0x13c58ac90>
filename = 's3://coiled-runtime-ci/test-scratch/test_nwm-9d96d682af1a4ba28a0b596c49dea501', mode = 'w', mmap = None, version = 2
maskandscale = False

    def __init__(self, filename, mode='r', mmap=None, version=1,
                 maskandscale=False):
        """Initialize netcdf_file from fileobj (str or file-like)."""
        if mode not in 'rwa':
            raise ValueError("Mode must be either 'r', 'w' or 'a'.")

        if hasattr(filename, 'seek'):  # file-like
            self.fp = filename
            self.filename = 'None'
            if mmap is None:
                mmap = False
            elif mmap and not hasattr(filename, 'fileno'):
                raise ValueError('Cannot use file object for mmap')
        else:  # maybe it's a string
            self.filename = filename
            omode = 'r+' if mode == 'a' else mode
>           self.fp = open(self.filename, '%sb' % omode)
E           FileNotFoundError: [Errno 2] No such file or directory: 's3://coiled-runtime-ci/test-scratch/test_nwm-9d96d682af1a4ba28a0b596c49dea501'

../../../mambaforge/envs/benchmarks/lib/python3.12/site-packages/scipy/io/_netcdf.py:247: FileNotFoundError
```

Reply:
Yes, but this isn't actually needed after all. It's writing out a small, in-memory Xarray dataset to a netcdf file. I'm fine calling that out of scope for this benchmark.